src/main/java/com/example/auctionapp/api/ItemController.java

```diff
@@ -0,0 +1,33 @@
+package com.example.auctionapp.api;
+
+import com.example.auctionapp.enumeration.ItemSort;
+import com.example.auctionapp.model.Item;
+import com.example.auctionapp.service.ItemService;
+import lombok.AllArgsConstructor;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Sort;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@CrossOrigin
+@RequestMapping("api/v1/items")
+@RestController
+@AllArgsConstructor
+public class ItemController {
+
+    @Autowired
+    ItemService itemService;
+
+    @GetMapping("/getall")
```

Comment:
do not specify getall in url, this can only be `/api/v1/items`.
and getItems method can be linked to `/api/v1/items/search` route